### DISCUSSION TITLE
Discard comments from comparison : handle python

### DISCUSSION THREAD

**Discussion by Bona72** [WinMerge/winmerge#1371] [Ideas]
Hi,
WinMerges has an option to discard comments from the comparison, which is very usefull.
It actually handles languages like C++ or Java, but does not handle python.
It would be usefull for many people I think that this would be done.
Thank you !
Bona

**Comment by sdottaka** [comment]
If you have a recent version of WinMerge, the ability to ignore comment differences should work on *.py files.

![image](https://user-images.githubusercontent.com/98126/174294439-c00d5853-bc20-4286-950a-c1d5d0dc3ef1.png)

**Comment by Bona72** [comment]
Thank you for your quick answer !
I am using WinMerge 2.16.18 on Windows 10. The menu is in french. Same behaviour in english.

![image](https://user-images.githubusercontent.com/97115851/174304630-58eddf60-ca27-48e4-9c7f-054d3237dfec.png)

**Comment by sdottaka** [comment]
In your case, WinMerge doesn't ignore comments well. However, it's not because it's a .py file, but because it's a limitation of WinMerge's capabilities. You will get the same result with .cpp and .java files.

The ability to ignore comments does not work on a line-by-line basis, but on a diff block basis. So if the only difference in the Diff block is the comment, it should work.

In your example, selecting the Whitespaces→Ignore all menu item as shown below should work.

![image](https://user-images.githubusercontent.com/98126/174309697-34bf4203-2968-49b8-bd15-54d6b09a9015.png)

**Comment by Bona72** [comment]
Indeed, it works when ignoring all whitespaces. It doesn't seem to me a consistent behaviour. If you do not want to see the differences in the comments, this should include whitespaces differences.
A future evolution ?
Anyway, thank you very much for your quick and accurate answers !
Bonne journée :)
